### ISSUE TITLE
Feature request: allow (automatic) printing of incoming messages

### ISSUE THREAD

**Issue by holtkamp** [telegramdesktop/tdesktop#2962]
**TL;DR:** it would be nice if messages from Telegram Desktop can be printed, possibly automatically for certain chats (with bots).

**Context**
The possibility of using Telegram Bots opens up interesting use cases. One of them that I am currently experimenting with is using Telegram as a medium to inform WebShop owners when an order has been received.

Some of the WebShop owners provided the feedback that it would be nice to directly print the incoming order confirmation. This is of particular use when the order involves food that needs to be prepared by "the kitchen", which use physical "receipts" to queue and process there work.

**Feature request**
That brings me to a feature request: allow to print messages received by Telegram Desktop. A next step would be to allow a user to configure Telegram Desktop to *automatically* print the messages that are received from a certain user / bot / in a certain chat.

**For the haters**
Of course one could argue that it might be better to stick to the good-old e-mail and simply print that. True, but:

1. Telegram offers two-way communication with a bot. So a WebShop owner could reply to an order confirmation with the expected preparation time, some questions about allergies and what not, etc.
2. A restaurant might have multiple devices (smartphone of the manager, tablet at the bar, PC + printer in the kitchen) to all keep track of the same stream of incoming orders...
3. etc...

Innovation++!

**Comment by auchri** [comment]
I think that this is not useful for the majority of users..you should use the api directly with your own program..There are enough libraries for every programming language.

**Comment by holtkamp** [comment]
@auchri , thanks for the swift response. 

>I think that this is not useful for the majority of users..

Yeah, could be, but is an assumption that should be weighed against the amount of work it is?

>you should use the api directly with your own program.
Yeah considered that, also considered submitting a PR for Telegram Desktop, too bad both are not my specialties... 😞

**Comment by stek29** [comment]
@holtkamp I doubt it would match with [feature philosophy](https://core.telegram.org/tsi/feature_philosophy).
> I am more into web development

[Telegram for Web :)](https://github.com/zhukov/webogram)

**Comment by holtkamp** [comment]
> I doubt it would match with feature philosophy.

Thanks, great read!
